t_frang/test_host_required.py

```diff
@@ -0,0 +1,583 @@
+
+    clients = clients
+
+    tempesta = tempesta
+
+    def setUp(self):
+        """Set up test."""
+        super().setUp()
+        self.klog = dmesg.DmesgFinder(ratelimited=False)
+
+    def test_h2_host_header_ok(self):
+        """Test with header `host`, success."""
+        curl = self.get_client("curl-1")
+
+        self.start_all_servers()
+        self.start_tempesta()
+
+        curl.start()
+        self.wait_while_busy(curl)
+
+        self.assertEqual(
+            CURL_CODE_OK,
+            curl.returncode,
+            ERROR_CURL.format(
+                str(curl.returncode),
+            ),
+        )
+        self.assertEqual(
+            self.klog.warn_count(WARN_IP_ADDR),
+            COUNT_WARNINGS_ZERO,
+            ERROR_MSG,
+        )
+        curl.stop()
+
+    def test_h2_empty_host_header(self):
+        """
+        Test with empty header `host`.
+
+        If there is no header `host`, curl set up it with ip address.
+        """
+        self._test_base_scenario(
+            curl_cli_id="curl-2", expected_warning=WARN_IP_ADDR, curl_code=CURL_CODE_OK
+        )
+
+    def test_h2_host_header_missing(self):
+        """Test with missing header `host`."""
+        self._test_base_scenario(
+            curl_cli_id="curl-3",
+            expected_warning=WARN_HEADER_MISSING,
+            curl_code=CURL_CODE_OK,
+        )
+
+    def test_h2_host_header_mismatch(self):
+        """Test with mismatched header `host`."""
+        self._test_base_scenario(
+            curl_cli_id="curl-4",
+            expected_warning=WARN_HEADER_MISMATCH,
+            curl_code=CURL_CODE_OK,
+        )
+
+    def test_h2_host_header_as_ip(self):
+        """Test with header `host` as ip address."""
+        self._test_base_scenario(
+            curl_cli_id="curl-5",
+            expected_warning=WARN_IP_ADDR,
+            curl_code=CURL_CODE_OK,
+        )
+
+    def test_h2_host_header_as_ipv6(self):
+        """Test with header `host` as ip v6 address."""
+        self._test_base_scenario(
+            curl_cli_id="curl-6",
+            expected_warning=WARN_HEADER_MISMATCH,
+            curl_code=CURL_CODE_OK,
+        )
+
+    def test_h2_host_header_forwarded(self):
+        """Test with mismsth header `forwarded`."""
+        self._test_base_scenario(
+            curl_cli_id="curl-7",
+            expected_warning=WARN_HEADER_FORWARDED2,
+            curl_code=CURL_CODE_OK,
+        )
+
+    def test_h2_host_header_double_forwarded(self):
+        """Test with double header `forwarded`."""
+        self._test_base_scenario(
+            curl_cli_id="curl-8",
+            expected_warning=WARN_HEADER_FORWARDED2,
+            curl_code=CURL_CODE_OK,
+        )
+
+    def test_h2_host_header_authority(self):
+        """Test with header `authority`."""
+        self._test_base_scenario(
+            curl_cli_id="curl-9",
+            expected_warning=WARN_HEADER_FORWARDED2,
+            curl_code=CURL_CODE_OK,
+        )
```

Comment:
Current `test_h2_host_header_authority` is not valid. You can't override `:authority` pseudo-header in such way via curl.

Reply:
ok, I will edit it here https://github.com/tempesta-tech/tempesta/issues/1717